### PR TITLE
Embeddings: collect and log stats for generating embeddings

### PR DESCRIPTION
**Pull request by camdencheek**
This collects and reports (currently via a log message) a set of statistics about building the embeddings index.

As a followup, this should get added to the database so that it can be exposed via the admin UI.

Stacked on #51021 

## Test plan

Added basic tests capturing the behavior of stats. Manually tested that the log message is readable and useful.

### CODE REVIEW

File: enterprise/internal/embeddings/embed/files.go [hunk 1]
```diff
@@ -74,19 +74,89 @@ func isExcludedFilePath(filePath string, excludedFilePathPatterns []*paths.GlobP
...
+
+	// File was excluded by configuration rules
+	SkipReasonExcluded
+
+	// File was excluded because we hit the max embedding limit for the repo
+	SkipReasonMaxEmbeddings
 )
 
+func (s SkipReason) String() string {
+	switch s {
```
Comment (jtibshirani) [inline enterprise/internal/embeddings/embed/files.go:100]: Could `SkipReason` just be a `string` (similar to `SkippedReason` in the search API)?

**Review by jtibshirani — APPROVED** [review]
Looks great, this will be very useful.

File: enterprise/internal/embeddings/embed/files.go (hunk 1 above)
Comment (camdencheek) [reply enterprise/internal/embeddings/embed/files.go:100]: Good idea!

**Review by eseliger — APPROVED** [review]
good stuff!

**Pull request merged**